### ISSUE TITLE
istioctl get gateways/policies not working

### ISSUE THREAD

**Issue by mbanikazemi** [istio/istio#5393]
Executing `istioctl get` suggests the following:
specify the type of resource to get. Types are routerule, egressrule, destinationpolicy, httpapispec, httpapispecbinding, quotaspec, quotaspecbinding, servicerole, servicerolebinding, virtualservice, gateway, serviceentry, destinationrule, policy

From the above list, `istioctl get` does not work properly for gateway and policy resources.

```
$ istioctl get gateways
...
Error: "gateways" not recognized. Please use non-hyphenated resource name "gateways"
```
```
$ istioctl get policies
...
Error: "policies" not recognized. Please use non-hyphenated resource name "policies"
```

**Comment by cmluciano** [comment]
strange that this works with kubectl 

```
kubectl get gateways       
NAME              AGE
httpbin-gateway   8m
```

**Comment by linsun** [comment]
@ZackButcher mentioned this is caused by the CRD that is used by istioctl is out of sync with the CRD used by k8s.  the release call agrees this is a release blocker.

**Comment by esnible** [comment]
Fixed by https://github.com/istio/istio/pull/5395

**Comment by mchendil82** [comment]
I am also facing the same issue .. I am able to create egress gateway rules.. but neither not able to delete or get it...  is this fixed in 0.8...? I am facing the following error:

_[root@xxxx bin]# ./istioctl get gateway
Usage:
  istioctl get <type> [<name>] [flags]

Examples:
# List all route rules
istioctl get routerules

Error: Please use non-hyphenated resource name "gateway"_

**Comment by esnible** [comment]
This is fixed in 0.8 dailies.  You only need to dow load the CLI.
